include/flash_map.h

```diff
@@ -33,47 +46,128 @@ extern "C" {
 #include <stddef.h>
 #include <sys/types.h>
 
-#define SOC_FLASH_0_ID 0	/* device_id for SoC flash memory driver */
-#define SPI_FLASH_0_ID 1	/* device_id for external SPI flash driver */
+#define SOC_FLASH_0_ID 0	/** device_id for SoC flash memory driver   */
+#define SPI_FLASH_0_ID 1	/** device_id for external SPI flash driver */
 
+/**
+ * @brief Structure for store flash partition data
+ *
+ * It is used as the flash_map array entry or stand-alone user data. Structure
+ * contains all data needed to operate on the flash partitions.
+ */
 struct flash_area {
-	u8_t fa_id;
+	u8_t fa_id; /** ID of flash area */
 	u8_t fa_device_id;
 	u16_t pad16;
-	off_t fa_off;
-	size_t fa_size;
-	const char *fa_dev_name;
+	off_t fa_off; /** flash partition offset */
+	size_t fa_size; /** flash partition size */
+	const char *fa_dev_name; /** flash device name */
 };
 
+/**
+ * @brief Structure for transfer flash sector boundaries
+ *
+ * This temlate is use for presentation of flash memory structure. It consume
+ * much less RAM than @ref flash_area
+ */
 struct flash_sector {
-	off_t fs_off;
-	size_t fs_size;
+	off_t fs_off; /** flash sector offset */
+	size_t fs_size; /** flash sector size */
 };
 
-/*
- * Start using flash area.
+/**
+ * @brief Retrieve partitions flash area form the flash_mpa.
+ *
+ * Function Retrieves flash_area form flash_map for given partition.
+ *
+ * @param[in] id ID of the flash partition.
+ * @param[out] flash_area Pointer which has to refernce flash_area. If
+ * @p ID is unknown, it will be NULL on output.
  */
 int flash_area_open(u8_t id, const struct flash_area **fa);
 
+/**
+ * @brief Close flash_area
+ *
+ * Reserved for future usage and external projects compatibility reason.
+ * Currently is NOP.
+ *
+ * @param[in] flash_area Flash area to be closed.
```

Comment:
The parameter name is ``fa`` not ``flash_area``, so doxygen is complaining about missing API documentation (lots of times).  See the messages in the shippable log (on the conversatation tab of this PR) for the doc build for this error and others that need fixing in the doxygen comments.